modules/navier_stokes/src/fvkernels/PINSFVEnergyTimeDerivative.C

```diff
@@ -59,10 +72,16 @@ PINSFVEnergyTimeDerivative::computeQpResidual()
   {
     auto elem_arg = makeElemArg(_current_elem);
     const auto state = determineState();
-    auto time_derivative = _rho(elem_arg, state) * _cp(elem_arg, state) * _var.dot(elem_arg, state);
-    if (_rho_dot)
-      time_derivative +=
-          (*_rho_dot)(elem_arg, state) * _cp(elem_arg, state) * _var(elem_arg, state);
+    ADReal time_derivative;
+    if (_h_dot)
+      time_derivative = _rho(elem_arg, state) * (*_h_dot)(elem_arg, state);
+    else
+    {
+      time_derivative = _rho(elem_arg, state) * (*_cp)(elem_arg, state) * _var.dot(elem_arg, state);
+      if (_rho_dot)
```

Comment:
In one branch you handle a time varying density, but in another you don't. It makes me wonder whether we should do it all in this PINSFV kernel or have a PWCNSFV extension. In `INSFVEnergyAdvection` you get no `rho_dot` handling

Reply:
ah good catch.
This could probably just become PWCNSFV ?

Reply:
Yea probably could. Although we're kind of in a hard spot with names. A lot of INSFV objects are appropriate for WCNSFV simulations so maybe we should rename all of those objects since I would regard WCNSFV as covering INSFV but not the other way around(?)

Reply:
anyway I think what you did in [af75916](https://github.com/idaholab/moose/pull/25922/commits/af75916433a5ea7c09aa88f31965b62b2d948312) addresses this particular conversation

Reply:
I m more ok with WCNSFV covering INSFV than INSFV covering WCNSFV and both are better than
INSFV sometimes covering WCNSFV sometimes not